src/content-tags-drawer/ContentTagsCollapsible.jsx

```diff
@@ -17,10 +18,143 @@ import ContentTagsDropDownSelector from './ContentTagsDropDownSelector';
 
 import ContentTagsTree from './ContentTagsTree';
 
+import { useContentTaxonomyTagsMutation } from './data/apiHooks';
+
+/**
+ * Util function that consolidates two tag trees into one, sorting the keys in
+ * alphabetical order.
+ *
+ * @param {object} tree1 - first tag tree
+ * @param {object} tree2 - second tag tree
+ * @returns {object} merged tree containing both tree1 and tree2
+ */
+const mergeTrees = (tree1, tree2) => {
+  const mergedTree = { ...tree1 };
+
+  const sortKeysAlphabetically = (obj) => {
+    const sortedObj = {};
+    Object.keys(obj)
+      .sort()
+      .forEach((key) => {
+        sortedObj[key] = obj[key];
+        if (obj[key] && typeof obj[key] === 'object') {
+          sortedObj[key].children = sortKeysAlphabetically(obj[key].children);
+        }
+      });
+    return sortedObj;
+  };
+
+  const mergeRecursively = (destination, source) => {
+    Object.entries(source).forEach(([key, sourceValue]) => {
+      const destinationValue = destination[key];
+
+      if (destinationValue && sourceValue && typeof destinationValue === 'object' && typeof sourceValue === 'object') {
+        mergeRecursively(destinationValue, sourceValue);
+      } else {
+        // eslint-disable-next-line no-param-reassign
```

Comment:
@bradenmacdonald Thanks for catching this very intricate bug! Though the main issue wasn't the reassigning of the param, as the linter mentions, since the `destination` is always copied over (each level of the recursion), however the issue was that the `source`  tree/value was not, hence it was getting modified.

I tried copying over the source value (tree) in the same style (using `{ ...sourceValue }`) however that made it really slow. I realize we have `lodash` installed, and it has a much more efficient clone method, so I used that instead, and it works great, the input objects no longer get modified. https://github.com/openedx/frontend-app-course-authoring/pull/704/commits/9fd8c6e40fe7ffc4c7895354cd02c32c5d84cf07